rotkehlchen/tests/app_upgrades/test_upgrades.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+from rotkehlchen.app_upgrades.manager import RotkiUpgradeManager
+from rotkehlchen.constants.assets import A_ETH, A_BTC
+from rotkehlchen.exchanges.data_structures import Trade
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.exchanges import check_saved_events_for_exchange
+from rotkehlchen.tests.utils.exchanges import create_test_binance
+from rotkehlchen.typing import (
+    Location,
+    AssetAmount,
+    Fee,
+    Price,
+    Timestamp,
+    TradeType,
+)
+
+
+@pytest.mark.parametrize('added_exchanges', [(Location.BINANCE, Location.POLONIEX, Location.KRAKEN)])  # noqa: E501
+def test_upgrade_1(rotkehlchen_api_server_with_exchanges, added_exchanges):
+    """
+    Test that the first app upgrade for rotki works. This update removes information about some
+    exchanges when there is more than one instance or if it is kraken.
+
+    In the test we setup instances of the exchanges to trigger the updates and one exchange
+    (POLONIEX) that shouldn't be affected.
+    """
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+    db = rotki.data.db
+    # Since we have initialized the rotkehlchen object the migration has already run
+    # and the setting has been updated. Let's reset this setting to what the user would experience
+    db.conn.cursor().execute(
+        'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
+        ('last_app_upgrade', 0),
+    )
```

Comment:
I suggest you make a mock fixture or a mock function to prepare the rotkehlchen object to the version you want to test like we go in `test_db_upgrades.py` with `_init_db_with_target_version`